generators/upgrade/index.js

```diff
@@ -151,57 +148,55 @@ module.exports = UpgradeGenerator.extend({
         },
 
         updateJhipster: function() {
-            this.log('Updating ' + GENERATOR_JHIPSTER + '...');
+            this.log('Updating ' + GENERATOR_JHIPSTER + '. This might take some time...');
             var done = this.async();
-            shelljs.exec('npm update -g ' + GENERATOR_JHIPSTER, {silent:true}, function (code, stdout, stderr) {
-                this.log(chalk.green('Updated ' + GENERATOR_JHIPSTER + ' to version ' + this.latestVersion));
+            shelljs.exec('npm install -g ' + GENERATOR_JHIPSTER, {silent:true}, function (code, stdout, stderr) {
+                if (code === 0) this.log(chalk.green('Updated ' + GENERATOR_JHIPSTER + ' to version ' + this.latestVersion));
+                else this.error('Something went wrong while updating generator! ' + stderr);
                 done();
             }.bind(this));
         },
 
         cleanUp: function() {
             var done = this.async();
-            this.git.exec('rm', {q: true}, ['-rf', '--ignore-unmatch', '*'], function(err, msg) {
-                if (err != null) return this.error('Unable to cleanup directory:\n' + err);
-            }.bind(this));
-            shelljs.exec('ls -a1 | grep -E -v \'.yo-rc.json|.git\|node_modules\' | xargs rm -rf +', {silent:true},
-                function (code, stdout, stderr) {
-                    this.log('Cleaned up directory');
-                    done();
-                }.bind(this));
+            shelljs.rm('-rf', '!(.yo-rc.json|.git)');
```

Comment:
maybe it's because I'm with linux, behavior might be a little different ; I'm going to accept all your changes and retest locally